src/middlewared/middlewared/plugins/network.py

```diff
@@ -1118,7 +1116,7 @@ async def do_create(self, data):
                         )
                 raise
         elif data['type'] == 'VLAN':
-            name = data.get('name') or f'vlan{data["vlan_tag"]}'
+            name = data.get('name') or await self.middleware.call('interface.get_next', 'vlan')
```

Comment:
What will happen if `data["name"]` is specified and points to already existing interface? Won't the same thing happen?

Reply:
On line `1054` we call `await self._common_validation` which has this logic:
```
1194         if 'name' in data and data['name'] in ifaces:
1195             verrors.add(f'{schema_name}.name', 'Interface name is already in use.')
```

so it should never reach this point?